java/memory/src/main/java/io/netty/buffer/UnsafeDirectLittleEndian.java

```diff
@@ -255,6 +258,28 @@ public boolean release(int decrement) {
   }
 
   @Override
+  public int setBytes(int index, InputStream in, int length) throws IOException {
```

Comment:
yes, Netty now relies on ```ByteBufAllocator.heapBuffer()``` to allocate byte arrays on heap, most likely to cache and reuse them if possible.